File: backend/vue/src/utils/jwt.ts
export interface JWTPayload {
  admin_id: number
  username: string
  role: string
  permissions: string
  exp?: number
}

export function parseJWT(token: string): JWTPayload | null {
  try {
    const payload = token.split('.')[1]
    const json = atob(payload.replace(/-/g, '+').replace(/_/g, '/'))
    return JSON.parse(json) as JWTPayload
  } catch {
    return null
  }
}

File: backend/go/internal/auth/jwt.go
package auth

import (
	"time"

	"github.com/golang-jwt/jwt/v5"
)

type Claims struct {
	AdminID     uint   `json:"admin_id"`
	Username    string `json:"username"`
	Role        string `json:"role"`
	Permissions string `json:"permissions"`
	jwt.RegisteredClaims
}

func GenerateToken(secret string, ttl time.Duration, adminID uint, username, role, permissions string) (string, error) {
	claims := Claims{
		AdminID:     adminID,
		Username:    username,
		Role:        role,
		Permissions: permissions,
		RegisteredClaims: jwt.RegisteredClaims{
			ExpiresAt: jwt.NewNumericDate(time.Now().Add(ttl)),
			IssuedAt:  jwt.NewNumericDate(time.Now()),
		},
	}
	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
	return token.SignedString([]byte(secret))
}

func ParseToken(secret, tokenStr string) (*Claims, error) {
	token, err := jwt.ParseWithClaims(tokenStr, &Claims{}, func(token *jwt.Token) (interface{}, error) {
		return []byte(secret), nil
	})
	if err != nil {
		return nil, err
	}
	if claims, ok := token.Claims.(*Claims); ok && token.Valid {
		return claims, nil
	}
	return nil, jwt.ErrTokenInvalidClaims
}
